### DISCUSSION TITLE
Fish shell support

### DISCUSSION THREAD

**Discussion by 2A2U** [binref/refinery#55] [Q&A] [answered]
does binary refinery supports fish shell?

Thanks.

**Comment by cxiao** [comment, marked as answer]
Hi! I use Fish shell with Binary Refinery. It works great for me, but there are a few things to be aware of:

- The `emit` command in Binary Refinery collides with the `emit` command in fish. Therefore, you may wish to install Binary Refinery with a prefix, as detailed in the README, or you may wish to alias `emit`. This issue has more details: https://github.com/binref/refinery/issues/22

- There are some special characters that Binary Refinery uses, that fish also uses. 

See the following information about special characters used in the Fish shell language:
https://fishshell.com/docs/current/language.html#escaping-characters:~:text=The%20special%20characters%20and%20their%20escape%20sequences%20are%3A

In practice the main one I've run into trouble with (like, once) is that curly braces, which are used by Binary Refinery are sometimes special in fish:
https://fishshell.com/docs/current/language.html#brace-expansion

**Reply by 2A2U** [reply]
Hi there,

My apologies for the delayed response. I'm glad to hear that Refinery supports the Fish shell. However, I encountered an issue when trying to install it with a prefix—the shell didn't recognize the prefix. Is there a specific way to install it with a prefix? I followed the instructions in the README, but it didn't work. Here are the steps I followed:

```bash
python -m venv binary-refinery
source binary-refinery/bin/activate.fish
set -x REFINERY_PREFIX r.
pip install -U binary-refinery
```

**Reply by huettenhain** [reply]
Out of curiosity, what is the result? Are there no shims installed at all, or does it indeed install the units without a prefix? Can you find anything on your machine named `r.emit` after the installation has concluded?

**Reply by huettenhain** [reply]
I also just found #22 and it seems like a prefix worked fine in that case. I am not sure why a prefix would not work in fish, there's very little magic happening here - the contents of the environment variable are simply prepended to the names of the command line shims that the refinery setup generates.

**Reply by 2A2U** [reply]
Hi My apologies for delayed response, i dont know what had happened by this time it works, I just did the same step and i works now. btw thanks for all of the answers
